File: Competetive programming/C++/Unit 2/cscoreboard.cpp
#include <iostream>
using namespace std;

int main() {

    int teams;

    cout << "Enter number of teams: ";
    cin >> teams;

    int solved[100], penalty[100];

    for(int i = 0; i < teams; i++) {

        cout << "\nTeam " << i + 1 << endl;

        cout << "Problems Solved: ";
        cin >> solved[i];

        cout << "Penalty Time: ";
        cin >> penalty[i];
    }

    cout << "\n--- Contest Scoreboard ---\n";

    for(int i = 0; i < teams; i++) {

        cout << "Team " << i + 1
             << " -> Solved: " << solved[i]
             << " Penalty: " << penalty[i]
             << endl;
    }

    return 0;
}

File: Competetive programming/Python/Unit 2/cscoreboard.py
print("===== CONTEST SCOREBOARD =====\n")

teams = int(input("Enter number of teams: "))

data = []

for i in range(teams):

    print(f"\nTeam {i+1}")

    solved = int(input("Solved Problems: "))
    penalty = int(input("Penalty: "))

    data.append([i+1, solved, penalty])

data.sort(key=lambda x: (-x[1], x[2]))

print("\nFINAL RANKING\n")

for rank, team in enumerate(data, start=1):

    print(
        rank,
        "Team",
        team[0],
        "Solved:",
        team[1],
        "Penalty:",
        team[2]
    )